udev/66-azure-ephemeral.rules

```diff
@@ -26,6 +34,19 @@ ENV{fabric_name}=="scsi0/lun0", ENV{fabric_name}="azure_root", GOTO="ci_azure_na
 ENV{fabric_name}=="scsi0/lun1", ENV{fabric_name}="azure_resource", GOTO="ci_azure_names"
 GOTO="cloud_init_end"
 
+LABEL="azure_nvme_datadisk"
+# azure data source ('DataSourceAzure.py') can't handle more than one resource disk
+# and it expects it to have a name /dev/disk/cloud/azure_resource
+# so we generate /dev/disk/cloud/azure_resource symlink for either disk 0 or disk 1
+# whichever we see first.
+ENV{disk_num}=="0", ENV{disk0_seen}="yes"
+ENV{disk_num}=="1", ENV{disk1_seen}="yes"
```

Comment:
I have not tested this part since I was never able to find a system where data disk is 0. Does the env var persist across invocations for all devices? I am not sure. If this does not work, we may need to use tmp files.